src/client.rs

```diff
@@ -215,14 +208,17 @@ pub async fn run(config: ClientConfig) -> Result<()> {
     // TODO: make sure it actually fails the bot if it exits.
     tokio::spawn(async move {
         if let Err(e) = db_connection.await {
-            eprintln!("connection error: {}", e);
+            panic!("connection error: {}", e);
         }
     });
 
     crate::migrations::runner()
         .run_async(&mut db_client)
         .await?;
 
+    let (plugin_senders, plugin_tasks): (Vec<_>, Vec<_>) =
+        crate::plugin::load(&config)?.into_iter().unzip();
```

Comment:
While really fancy, IMO this adds an unnecessary bit of cognitive load. Can you instead have `PluginMeta` be a struct below and then do something like this:

```rust
let mut plugin_senders = Vec::new();
let mut plugin_tasks = Vec::new();
for meta in crate::plugin::load(&config)?.into_iter() {
    plugin_senders.push(meta.sender);
    plugin_tasks.push(meta.task);
}
```

This makes what you're doing much clearer.